test/stdlib/StringAPI.swift

```diff
@@ -397,6 +397,105 @@ StringTests.test("Regression/rdar-33276845") {
   }
 }
 
+StringTests.test("Regression/corelibs-foundation") {
+  struct NSRange { var location, length: Int }
+  
+  func NSFakeRange(_ location: Int, _ length: Int) -> NSRange {
+    return NSRange(location: location, length: length)
+  }
+
+  func substring(of _storage: String, with range: NSRange) -> String {
+    let start = _storage.utf16.startIndex
+    let min = _storage.utf16.index(start, offsetBy: range.location)
+    let max = _storage.utf16.index(
+      start, offsetBy: range.location + range.length)
+
+    if let substr = String(_storage.utf16[min..<max]) {
+      return substr
+    }
+    //If we come here, then the range has created unpaired surrogates on either end.
+    //An unpaired surrogate is replaced by OXFFFD - the Unicode Replacement Character.
+    //The CRLF ("\r\n") sequence is also treated like a surrogate pair, but its constinuent
+    //characters "\r" and "\n" can exist outside the pair!
```

Comment:
Is this a compatibility-with-NSRange/NSString/CFString property, or should Swift String's UTF16View also honor these semantics?